types/react-redux/index.d.ts

```diff
@@ -49,13 +53,72 @@ export interface InferableComponentDecorator<TOwnProps> {
  */
 export declare function connect<TOwnProps>(): InferableComponentDecorator<TOwnProps>;
 
+export declare function connect<TStateProps, TOwnProps>(
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>
+): ComponentDecorator<TStateProps, void, TOwnProps>;
+
+export declare function connect<TDispatchProps, TOwnProps>(
+    mapStateToProps: undefined,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
+): ComponentDecorator<void, TDispatchProps, TOwnProps>;
+
 export declare function connect<TStateProps, TDispatchProps, TOwnProps>(
-    mapStateToProps?: MapStateToProps<TStateProps, TOwnProps> | MapStateToPropsFactory<TStateProps, TOwnProps>,
-    mapDispatchToProps?: MapDispatchToProps<TDispatchProps, TOwnProps> | MapDispatchToPropsFactory<TDispatchProps, TOwnProps>,
-    mergeProps?: MergeProps<TStateProps, TDispatchProps, TOwnProps>,
-    options?: Options
+    mapStateToProps: MapStateToPropsParam<TStateProps, TOwnProps>,
+    mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>
 ): ComponentDecorator<TStateProps, TDispatchProps, TOwnProps>;
 
+export declare function connect<TStateProps, TOwnProps, TMergedProps>(
```

Comment:
Dropping a type argument should be fine as long as you aren't explicitly type-hinting `connect` right now, but if you are, this could be an annoying and unexpected breaking update. Though I would like to reduce the number of type arguments, I think the principle of least surprise would ask that we don't and instead just leave unused type parameters around. However, I'm not sure how this play with `noUnusedLocals`/`noUnusedParameters`. :/